src/language-html/preprocess.js

```diff
@@ -313,7 +317,7 @@ function extractWhitespaces(ast /*, options*/) {
       node.children.length === 0 ||
       (node.children.length === 1 &&
         node.children[0].type === "text" &&
-        node.children[0].value.trim().length === 0)
+        htmlTrim(node.children[0].value).length === 0)
```

Comment:
Good job, `.trim()` is not safe in some cases, maybe we should look at `.trim()` in all code, we can have other problems, for example for `css`